src/components/Login.jsx

```diff
@@ -112,7 +177,7 @@ function Login({ setCredentials }) {
       } else if (response.status === 401) {
         window.alert("Login failed.\n\nThat was the wrong password.");
       } else if (response.status === 404) {
-        setCredentials(data);
+        setCredentials({ unregistered: true, ...data });
       } else if (response.status === 504) {
         window.alert(
```

Comment:
Consider DRY rule. Combine repeated alert calls into a single function call to avoid redundancy.

function showAlert(message) {
  window.alert(message);
}

Reply:
In this case, this would increase the code's complexity (by adding an extra function) without affecting the DRY principle.

```javascript
window.alert("One");
window.alert("Two");
window.alert("Three");
```
has the same number of statements as
```javascript
showAlert("One");
showAlert("Two");
showAlert("Three");
```